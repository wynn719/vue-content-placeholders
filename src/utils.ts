export function usePrefix(prefix: string) {
  return `vue-content-placeholders-${prefix}` as const
}
