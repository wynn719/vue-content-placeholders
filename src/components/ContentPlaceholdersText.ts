import { defineComponent, h } from 'vue-demi'
import { usePrefix } from '@/utils'

export default defineComponent({
  props: {
    lines: {
      type: Number,
      default: 4,
    },
  },
  setup(props) {
    const className = usePrefix('text')
    return () => {
      const lines = Array.from({ length: props.lines }).map((_, idx) => {
        return h('div', { key: idx, class: `${className}__line` })
      })

      return h('div', { class: className }, lines)
    }
  },
})
