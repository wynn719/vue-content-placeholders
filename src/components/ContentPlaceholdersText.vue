<template>
  <div :class="className">
    <div v-for="n in lines" :key="n" :class="`${className}__line`" />
  </div>
</template>

<script lang="ts" setup>
import { usePrefix } from '@/utils'

defineProps({
  lines: {
    type: Number,
    default: 4,
  },
})

const className = usePrefix('text')
</script>
