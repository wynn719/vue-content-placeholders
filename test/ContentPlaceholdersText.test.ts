import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import ContentPlaceholdersText from '@/components/ContentPlaceholdersText'
import { usePrefix } from '@/utils'

describe('ContentPlaceholders', () => {
  const createWrapper = (args?: any) => mount(ContentPlaceholdersText, {
    ...args,
  })

  it('should render correctly', () => {
    const wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support lines', () => {
    const lines = Math.floor(Math.random() * 20)
    const wrapper = createWrapper({
      props: {
        lines,
      },
    })
    expect(wrapper.findAll(`.${usePrefix('text__line')}`)).toHaveLength(lines)
  })
})
