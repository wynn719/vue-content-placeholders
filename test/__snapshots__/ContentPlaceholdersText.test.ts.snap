// Vitest Snapshot v1

exports[`ContentPlaceholders > should render correctly 1`] = `
"<div class=\\"vue-content-placeholders-text\\">
  <div class=\\"vue-content-placeholders-text__line\\"></div>
  <div class=\\"vue-content-placeholders-text__line\\"></div>
  <div class=\\"vue-content-placeholders-text__line\\"></div>
  <div class=\\"vue-content-placeholders-text__line\\"></div>
</div>"
`;
